### ISSUE TITLE
dvc: support http(s) as an external dependency

### ISSUE THREAD

**Issue by efiop** [iterative/dvc#1146]

**Comment by mroutis** [comment]
@efiop ,I didn't understand the second case;
When specifying a dependency with the `http` protocol should we try to make a GET request and cache the content?

**Comment by efiop** [comment]
No, it means that we should get ETAG(if available, if not -- throw an error) and save it instead of md5(how it is done in s3, gs drivers). No downloading should be done on dvc side. Notice that In the second example user downloads the file _himself_ and saves it into `data` file, which is cached by dvc. Maybe that example is a bit confusing, sorry. Imagine you have a script that uses the remote file without downloading it(e.g. it is way to big), then you can use `dvc run -d https://example.com/file -o processed python process.py` so that when remote file changes, your stage gets reproduced.
